diff --git a/gen_web.go b/gen_web.go
index 125b7abc..eb33a8d5 100644
--- a/gen_web.go
+++ b/gen_web.go
@@ -14,14 +14,16 @@ import (
 
 func main() {
 	dir, _ := os.Getwd()
-	os.Chdir("web")
-	if run("yarn") != nil {
-		os.Exit(1)
-	}
-	if run("yarn", "run", "build") != nil {
-		os.Exit(1)
+	if _, err := os.Stat("web/build/static"); os.IsNotExist(err) {
+		os.Chdir("web")
+		if run("yarn") != nil {
+			os.Exit(1)
+		}
+		if run("yarn", "run", "build") != nil {
+			os.Exit(1)
+		}
+		os.Chdir(dir)
 	}
-	os.Chdir(dir)
 
 	compileHtml := "web/build/"
 	srcGo := "server/web/pages/"
